### PR TITLE
feat(useDateFormat): support dd, ddd and dddd formatter

### PR DESCRIPTION
**Pull request by aki77**
### Description

Support `dd`, `ddd` and `dddd` format. #1478

### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [x] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/shared/useDateFormat/index.ts [hunk 1]
```diff
@@ -16,25 +23,28 @@ export const formatDate = (date: Date, formatStr: string) => {
...
-    HH: `${hours}`.padStart(2, '0'),
-    h: `${hours % 12 || 12}`.padStart(1, '0'),
-    hh: `${hours % 12 || 12}`.padStart(2, '0'),
-    m: String(minutes),
-    mm: `${minutes}`.padStart(2, '0'),
-    s: String(seconds),
-    ss: `${seconds}`.padStart(2, '0'),
-    SSS: `${milliseconds}`.padStart(3, '0'),
-    d: day,
+  const matches: Record<string, () => string | number> = {
```
Comment (aki77) [inline packages/shared/useDateFormat/index.ts:26]: Changed to function so as not to unnecessarily execute toLocaleDateString and degrade performance.

File: packages/shared/useDateFormat/index.ts [hunk 2]
```diff
@@ -4,10 +4,17 @@ import { computed } from 'vue-demi'
 
 export type DateLike = Date | number | string | undefined
 
+export interface UseDateFormatOptions {
+  /**
+   * A locale string
```
Comment (webfansplz) [inline packages/shared/useDateFormat/index.ts:9]: `The locale(s) to use`

**Comment by webfansplz** [comment]
LGTM, Can you add the usage to the docs (format with locales), Thanks !

**Comment by aki77** [comment]
@webfansplz  

Thanks for the review!
Added usage to the docs.

**Review by webfansplz — APPROVED** [review]

**Comment by aki77** [comment]
@antfu @webfansplz Thanks for the review!
What else do I need to do to get this PR merged?

**Pull request merged**